### PR TITLE
Add short data.table intro

### PR DESCRIPTION
**Pull request by mllg**

### CODE REVIEW

File: book/02-basics-data.table.qmd [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+First, the index operator `[` works slightly different.
+For both objects, the first argument (`i`) is used to select rows and the second argument (`j`) is used to select columns.
+However, column names are in the search path, and thus can be used directly:
+```{r}
+df[df$y == "a", ]
+dt[y == "a", ]
+```
+
+Second, there is no optional `drop` argument (`drop` is always `FALSE` for `data.table()`), but instead multiple additional arguments to query the data from the `data.table()` in a very concise way.
+Most importantly, you can group the data with `by` and combine this with aggregating functions in `j`:
```
Comment (larskotthoff) [inline book/02-basics-data.table.qmd:32]: Not clear what "in `j`" means here (there's no `j` in the code).

Comment (mllg) [reply book/02-basics-data.table.qmd]: yep, clarified this

**Review by pat-s — CHANGES_REQUESTED** [review]
- [ ] update chunk naming
- [ ] empty lines between code chunks and subsequent/previous text

File: book/02-basics-data.table.qmd [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+For both objects, the first argument (`i`) is used to select rows and the second argument (`j`) is used to select columns.
+However, column names are in the search path, and thus can be used directly:
+
+```{r 02-basics-data.table-003}
+df[df$y == "a", ]
+dt[y == "a", ]
+```
+
+Second, there is no optional `drop` argument (`drop` is always `FALSE` for `data.table()`), but instead multiple additional arguments to query the data from the `data.table()` in a very concise way.
+Most importantly, you can group the data with argument `by` and combine this with aggregating functions provided via the second argument (`j`):
```
Comment (larskotthoff) [inline book/02-basics-data.table.qmd:35]: Unclear what `j` is here -- `y` in the example?

**Pull request merged**